canonical-web-and-design/maas-ui
# refactor(controllers): update controllers to React Router v6

## Done

- Update controllers to support React Router v6.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the list of controllers. It should appear.
- Open your redux devtools  and then click on a controller. The controller details should appear and there should be `controller/get` and `controller/setActive` dispatches.
- Click on each controller tab. The tabs should appear and there should not be any more get or setActive dispatches.
- Go to the summary tab and scroll down until the "See storage" link is at the top of your screen.
- Click the "See storage" link and you should be taken to the storage tab and you should be scrolled to the top of your screen.

## Fixes

Fixes: #4197.
